master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java

```diff
@@ -135,6 +135,17 @@ public void updateWorkerRemoveMeta(
     excludedWorkers.remove(worker);
   }
 
+  public void removeWorkersUnavailableInfoMeta(List<WorkerInfo> unavailableWorkers) {
+    synchronized (workers) {
+      for (WorkerInfo workerInfo : unavailableWorkers) {
+        if (lostWorkers.containsKey(workerInfo)) {
+          lostWorkers.remove(workerInfo);
+          shutdownWorkers.remove(workerInfo);
```

Comment:
`shutdownWorkers` do not necessarily overlap with `lostWorkers`. Need to check `shutdownWorkers` independently, or don't check contains at all.

Reply:
> `shutdownWorkers` do not necessarily overlap with `lostWorkers`. Need to check `shutdownWorkers` independently, or don't check contains at all.

Only those lost workers which also in shutdownWorkers need remove.

Reply:
Right, thanks :)